### PR TITLE
Define a Dockerfile for local development purposes

### PR DESCRIPTION
**Pull request by rayanht**
#18

### CODE REVIEW

**Comment by rayanht** [comment]
An alternative to this Dockerfile could simply run the `quick_start.sh` but my motivation is that this is a lot more clearer and maintanable.

**Comment by rayanht** [comment]
@googlebot I signed it!

**Comment by JonZeolla** [comment]
This cannot work as currently documented because `127.0.0.1` in the container does not refer to `127.0.0.1` on the host.  Consider setting up a docker network, or something with compose.  The docker network approach would be something like:
```bash
docker network create example
docker run --network example --name unauthenticated-jupyter-notebook -p 8888:8888 -d jupyter/base-notebook start-notebook.sh --NotebookApp.token=''
docker run --network example tsunami java -cp "/usr/tsunami/tsunami-main-0.0.2-SNAPSHOT-cli.jar:/usr/tsunami/plugins/*" -Dtsunami-config.location=/usr/tsunami/tsunami.yaml com.google.tsunami.main.cli.TsunamiCli --hostname-target=unauthenticated-jupyter-notebook
```

Minor other feedback:
- Suggest updating the `TARGET_IP` approach to differentiate between v4 and v6 and adding support for `--hostname-target` (see [this](https://github.com/google/tsunami-security-scanner/blob/6314ba862884833be797e7e40d97c5b2cfc32101/main/src/main/java/com/google/tsunami/main/cli/option/ScanTargetCliOptions.java#L32-L45))
- Consider using a combination of `ENTRYPOINT` and `CMD` to allow arguments to be overridden to the defined base command at `docker run`.  Something like the following would allow a simplification of the docker network example's last command to `docker run --network example tsunami -Dtsunami-config.location=/usr/tsunami/tsunami.yaml com.google.tsunami.main.cli.TsunamiCli --hostname-target=unauthenticated-jupyter-notebook`
    ```
    ENTRYPOINT ["java", "-cp", "/usr/tsunami/tsunami-main-0.0.2-SNAPSHOT-cli.jar:/usr/tsunami/plugins/*"]
    CMD -Dtsunami-config.location=/usr/tsunami/tsunami.yaml com.google.tsunami.main.cli.TsunamiCli --ip-v4-target=$TARGET_IP 
    ```

Note:  I'm not affiliated with this project or Google, just giving some feedback...

**Comment by rayanht** [comment]
@JonZeolla Thanks for the suggestions! Re: the docker network, how about simply running the Tsunami image with `--network="host"` ? This would allow to scan for vulnerabilities in applications that aren't necessarily docker images themselves.

**Comment by vdjagilev** [comment]
Have a little side note for this one.

I am not sure how `gradle` defines a version for any particular `.jar` file it creates. But in this case `tsunami-main-0.0.2-SNAPSHOT-cli.jar` is hardcoded into Dockerfile. I think it would be good if there is no need to change `Dockerfile` each time when jar file name is different. 

Maybe it's possible to do something like this `JARFILE=$(find ...)`, I think this would be good example https://github.com/google/tsunami-security-scanner/blob/master/quick_start.sh#L57

**Comment by rayanht** [comment]
@vdjagilev I considered it but I wasn't aware of the `basename` command. Will definitely amend, thank you for the heads up!

**Comment by rayanht** [comment]
A bit of an update on this, there is no way to persist variables between RUN commands in Docker and I can't seem to get string interpolation to behave nicely inside the ENTRYPOINT. 

To be honest, I don't really see the point in specifying the version number in the filename of the generated jar. Could we perhaps drop it and have a standard filename?

**Comment by JonZeolla** [comment]
How I've solved this in the past is to use a build arg (`ARG`), and then use something like a `Makefile` or wrapper shell script that grabs the version [from the project](https://github.com/google/tsunami-security-scanner/blob/6314ba862884833be797e7e40d97c5b2cfc32101/build.gradle#L16) and passes it into the `docker build` via an arg.

While its a completely different project and context, here is a [concrete example](https://github.com/SeisoLLC/easy_sast/blob/master/Makefile#L46-L47)

**Comment by rayanht** [comment]
@JonZeolla Yep that would definitely work, it's just that I would've preferred a fully self-contained Docker image that the end user could simply `docker run`

**Comment by JonZeolla** [comment]
What about just finding the jar at build time and rename it to a standard name like `tsunami-main-cli.jar` and use that in the `ENTRYPOINT`?

**Comment by rayanht** [comment]
@JonZeolla ...it's so obvious that I didn't think about it 😅

**Comment by vdjagilev** [comment]
Maybe it's better to leave old documentation and divide installation approaches into 2 topics (with docker and without docker)? Use case: Running docker in VM with Kali Linux can be memory-expensive

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+RUN cp $(find "./" -name 'tsunami-main-*-cli.jar') /usr/tsunami/tsunami.jar
+RUN cp ./tsunami.yaml /usr/tsunami
+
+WORKDIR /usr/tsunami
+
+RUN mkdir logs/
+
+ENTRYPOINT ["java", "-cp", "tsunami.jar:plugins/*", "-Dtsunami-config.location=tsunami.yaml", "com.google.tsunami.main.cli.TsunamiCli"]
+CMD ["--ip-v4-target=127.0.0.1", "--scan-results-local-output-format=JSON", "--scan-results-local-output-filename=logs/tsunami-output.json"]
```
Comment (magl0) [inline Dockerfile:34]: nit: per code style, please add a new line at the end of this file.

**Review by magl0 — CHANGES_REQUESTED** [review]

**Pull request merged**